docs/docs/k8s/ingress.md

```diff
@@ -0,0 +1,363 @@
+| `ingress.pomerium.io/tls_custom_ca_secret`            | Name of Kubernetes `tls` Secret containing a custom [CA certificate][`tls_custom_ca_secret`] for the upstream. |
+| `ingress.pomerium.io/tls_client_secret`               | Name of Kubernetes `tls` Secret containing a [client certificate][`tls_client_secret`] for connecting to the upstream. |
+| `ingress.pomerium.io/tls_downstream_client_ca_secret` | Name of Kubernetes `tls` Secret containing a [Client CA][client-certificate-authority] for validating downstream clients. |
+| `ingress.pomerium.io/secure_upstream`                 | When set to true, use `https` when connecting to the upstream endpoint. |
+
+::: tip
+
+Every value for the annotations above must be in `string` format.
+
+:::
+
+### Cert Manager Integration
+
+Pomerium Ingress Controller supports utilizing [cert-manager](https://cert-manager.io/) for provisioning certificates.  These may come from the [`Ingress` shim](https://cert-manager.io/docs/usage/ingress/) or explicitly configured [`Certificate` resources](https://cert-manager.io/docs/usage/certificate/).
+
+To use [HTTP01 Challenges](https://cert-manager.io/docs/configuration/acme/http01/) with your Issuer, configure the solver class to match the Ingress Controller.  The Ingress Controller will automatically configure policy to facilitate the HTTP01 challenge:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: example-issuer
+spec:
+  acme:
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    privateKeySecretRef:
+      name: example-issuer-account-key
+    solvers:
+    - http01:
+       ingress:
+         class: pomerium
+```
+
+An example of using the Ingress Shim with an Ingress resource managed by Pomerium:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/issuer: example-issuer
+    ingress.pomerium.io/policy: '[{"allow":{"and":[{"email":{"is":"user@exampledomain.com"}}]}}]'
+  name: example
+spec:
+  ingressClassName: pomerium
+  rules:
+  - host: example.localhost.pomerium.io
+    http:
+      paths:
+      - backend:
+          service:
+            name: example
+            port:
+              name: http
+        path: /
+        pathType: Prefix
+  tls:
+  - hosts:
+    - example.localhost.pomerium.io
+    secretName: example-tls
+```
+
+## HTTPS endpoints
+
+The `Ingress` spec assumes that all communications to the service is sent in plaintext. For more information, see the [TLS](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls) section of the Ingress API documentation. Pomerium supports HTTPS communication with upstream endpoints, including mTLS.
+
+Annotate your `Ingress` with
+
+```yaml
+ingress.pomerium.io/secure_upstream: true
+```
+
+Additional TLS may be supplied by creating a Kubernetes secret(s) in the same namespaces as `Ingress` resource. Please note that we do not support file paths or embedded secret references.
```

Comment:
```suggestion
Additional TLS certificates may be supplied by creating a Kubernetes secret(s) in the same namespaces as `Ingress` resource. Please note that we do not support file paths or embedded secret references.
```

Certificates? CAs? Configurations?

Reply:
I'm pretty sure this is just for certificates, and stands as is. @travisgroth to confirm.

Reply:
👍  yes